text/1021-rewrites.rst

```diff
@@ -0,0 +1,210 @@
+::
+
+    Status: Draft
+    Type: Feature
+    Created: 2023-01-12
+    Authors: Michel J. Sullivan <sully@msully.net>
+
+===========================
+RFC 1021: Mutation rewrites
+===========================
+
+This RFC proposes a mechanism for specifying rewrite rules that are
+applied to properties and links on INSERT and UPDATE of objects.
+
+They can be thought of as a generalization of ``default``.
+
+Motivation
+==========
+
+It is sometimes useful to have a pointer in an object automatically updated
+whenever the object has an ``UPDATE`` performed on it.
+
+The most straightforward example of this is having a
+``modification_time`` field that tracks when an object was last
+modified.
+
+It is also valuable when (probably for performance reasons) you wish
+to maintain a field that is derived from other object
+state. (For example, maintaining a count of friends.)
+
+Note
+====
+
+This is a companion to
+`RFC 1020 <https://github.com/edgedb/rfcs/blob/master/text/1020-triggers.rst>`_,
+in that they work together to cover most of the desired use cases that motivated
+a previous `previous version triggers RFC <https://github.com/edgedb/rfcs/pull/70>`_
+
+Specification
+=============
+
+Synopsis::
+
+    type <type-name> "{"
+        { property | link } <name> -> <type> "{"
+            rewrite { insert | update } [, ... ] using (<expr>)
+        "}"
+    "}"
+
+We allow pointer definitions to be augmented with a ``rewrite`` rule.
+
+Expression interpretation
+-------------------------
+In ``<expr>``, ``__subject__`` (and the implicit path prefix) will be
+bound to the new value object being inserted or updated.
```

Comment:
Is referencing links, links properties or properties of link targets allowed?

Reply:
Yeah I think so